pkg/scaling/executor/scale_scaledobjects.go

```diff
@@ -16,18 +19,38 @@ func (e *scaleExecutor) RequestScale(ctx context.Context, scaledObject *kedav1al
 		"scaledObject.Namespace", scaledObject.Namespace,
 		"scaleTarget.Name", scaledObject.Spec.ScaleTargetRef.Name)
 
-	currentScale, err := e.getScaleTargetScale(ctx, scaledObject)
-	if err != nil {
-		logger.Error(err, "Error getting information on the current Scale (ie. replias count) on the scaleTarget")
-		return
+	// Get the current replica count. As a special case, Deployments fetch directly from the object so they can use the informer cache
+	// to reduce API calls. Everything else uses the scale subresource.
+	var currentScale *autoscalingv1.Scale
+	var currentReplicas int32
+	targetRef := scaledObject.Spec.ScaleTargetRef
+	// TODO Should this use a more generic runtime.Object approach using scheme.Scheme.New()?
+	if (targetRef.APIVersion == "" || strings.HasPrefix(targetRef.APIVersion, "apps/")) &&
```

Comment:
This condition could be simpler if we use `scaledObject.status.ScaleTargetGVKR` https://github.com/kedacore/keda/blob/5819a1ffdde9c9f14efe74eee783dd34718de363/api/v1alpha1/scaledobject_types.go#L88

This should have the correct API, Version and Kind resolved, see:
https://github.com/kedacore/keda/blob/5819a1ffdde9c9f14efe74eee783dd34718de363/pkg/util/gvkr.go#L18

Reply:
And we could handle `StatefulSet` directly as well.

Reply:
Added both, much cleaner!